src/constraint/spec.ts

```diff
@@ -719,12 +719,27 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
         const xEncQ = specM.getEncodingQueryByChannel('x');
         const yEncQ = specM.getEncodingQueryByChannel('y');
 
-        // TODO(#186): take non-spatial channel into account
         if ((!isFieldQuery(xEncQ) || isDiscrete(xEncQ)) &&
-          (!isFieldQuery(yEncQ) || isDiscrete(yEncQ)) &&
-          !specM.isAggregate() // TODO: refactor based on statistics
-        ) {
-          return false;
+            (!isFieldQuery(yEncQ) || isDiscrete(yEncQ))) {
+          if (!specM.isAggregate()) {
+            return false;
+          } else {
+            return every(specM.getEncodings(), (encQ) => {
+              let channel = encQ.channel;
+
+              if (channel === Channel.COLOR || channel === Channel.DETAIL ||
```

Comment:
Instead checking for all non-xy channel, check if they are not x/y/row/column. 

(The number of non-position can grow in the future and manually checking one by one like this will make this constraint easily outdated.)